[DOC]Remove outdated compatibility notices and rework cloud notice (#938)
logstash-plugins/logstash-output-elasticsearch: docs/index.asciidoc

==== Description

> If you plan to use the Kibana web interface, use the Elasticsearch output plugin

Reviewer: ```suggestion
If you plan to use the Kibana web interface to analyze data transformed by Logstash, use the Elasticsearch output plugin
```

> to get your log data into Elasticsearch. 

Reviewer: ```suggestion
to get your data into Elasticsearch. 
```

> This output only speaks the HTTP protocol. HTTP is the preferred protocol for interacting with Elasticsearch.

Reviewer: We can remove this paragraph as node protocol hasn’t been available for some time and we don’t provide transport protocol support anymore in Logstash.

> This output only speaks the HTTP protocol. HTTP is the preferred protocol for interacting with Elasticsearch.
> We strongly encourage the use of HTTP over the node protocol for a number of reasons. HTTP is only marginally slower,
> yet far easier to administer and work with. When using the HTTP protocol one may upgrade Elasticsearch versions without having
> to upgrade Logstash in lock-step. 

Reviewer: ```suggestion
This output only speaks the HTTP protocol as it is the preferred protocol for interacting with Elasticsearch.
In previous versions it was possible to communicate with Elasticsearch through the transport protocol, which is now reserved for internal cluster communication between nodes https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-transport.html. This has been deprecated in Elasticsearch 7.0.0 and will be removed in 8.0.0 https://www.elastic.co/guide/en/elasticsearch/reference/current/java-clients.html
```

==== Description

If you plan to use the Kibana web interface to analyze data transformed by
Logstash, use the Elasticsearch output plugin to get your data into
Elasticsearch. 

This output only speaks the HTTP protocol as it is the preferred protocol for
interacting with Elasticsearch. In previous versions it was possible to
communicate with Elasticsearch through the transport protocol, which is now
reserved for internal cluster communication between nodes
https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-transport.html.

> This has been deprecated in Elasticsearch 7.0.0 and will be removed in 8.0.0

Author: @jsvd Thanks for adding this info. Can you clarify this?
"This" (in "This has been deprecated...") means  "transport protocol?"

Reviewer: Using the transport protocol to communicate with the cluster

Author: Thanks for clarifying. Please see what you think.
